bundle/edu.gemini.lchquery.servlet/src/main/java/edu/gemini/lchquery/servlet/LchQueryServlet.java

```diff
@@ -1,201 +0,0 @@
- * and targets for Semester 2012B with the given observation status and using
- * a LGS):</p>
- *
- * <p>
- * http://localhost:8296/lchquery?programSemester=2012B&programReference=GN*&observationStatus=Phase2|For Review|In Review|For Activation|On Hold|Ready|Ongoing*&observationAo=Altair + LGS
- * </p>
- *
- * <p>The program active and completed flags and the too status might be
- * important at a later stage, also the program title and observation name in
- * case they keep "marking" the engineering programs by using distinct
- * program and/or observation names.</p>
- *
- * <p>See LCH-63:</p>
-*/
-public final class LchQueryServlet extends HttpServlet {
-
-    private static final Logger LOG = Logger.getLogger(LchQueryServlet.class.getName());
-
-    private static final int ALL_OK = 200;
-    private static final int INVALID_REQUEST = 400;
-    private static final int SERVER_ERROR = 500;
-
-    private final IDBDatabaseService odb;
-    private final Set<Principal> user;
-
-    public LchQueryServlet(IDBDatabaseService odb, Set<Principal> user) {
-        this.odb = odb;
-        this.user = user;
-    }
-
-    /**
-     * Handles gets requests.
-     * @param request
-     * @param response
-     * @throws IOException
-     */
-    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
-        OutputStream out = response.getOutputStream();
-        BufferedOutputStream bos = new BufferedOutputStream(out);
-
-        String pathInfo = request.getPathInfo();
-        final LchQueryFunctor.QueryType queryType;
-        if (pathInfo.startsWith("/programs")) {
-            queryType = LchQueryFunctor.QueryType.PROGRAMS;
-        } else if (pathInfo.startsWith("/observations")) {
-            queryType = LchQueryFunctor.QueryType.OBSERVATIONS;
-        } else if (pathInfo.startsWith("/targets")) {
-            queryType = LchQueryFunctor.QueryType.TARGETS;
-        } else {
-            response.setStatus(INVALID_REQUEST);
-            response.setContentType("text/plain");
-            bos.write(("ERROR: Invalid query selector: " + pathInfo
-                    + ". Should start with /programs, /observations or /targets").getBytes());
-            bos.close();
-            return;
-        }
-
-        String programSemester = request.getParameter(LchQueryService.PARAMETER_PROGRAM_SEMESTER);
-        String programTitle = request.getParameter(LchQueryService.PARAMETER_PROGRAM_TITLE);
-        String programReference = request.getParameter(LchQueryService.PARAMETER_PROGRAM_REFERENCE);
-        String programActive = request.getParameter(LchQueryService.PARAMETER_PROGRAM_ACTIVE);
-        String programCompleted = request.getParameter(LchQueryService.PARAMETER_PROGRAM_COMPLETED);
-        String programNotifyPi = request.getParameter(LchQueryService.PARAMETER_PROGRAM_NOTIFY_PI);
-        String programRollover = request.getParameter(LchQueryService.PARAMETER_PROGRAM_ROLLOVER);
```

Comment:
Formerly, to extract parameters in `LchQueryServlet`, we had to deal with nonsense like this.